### PR TITLE
Fix #9818 - Windows documentation workflow is failing

### LINKED ISSUE #9818: Windows documentation workflow is failing
Issue overview
--------------

Fix the windows documentation workflow

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------
 - Fixes #9818 

The mem flags were passed as `'extra-mem-top=2000000 --extra-mem-bot=4000000'` which is apparently understood as one argument incorrectly in miktex 22.12

On 21.6 everything works, when I upgraded to 22.12 (like CI) it stopped working. I suppose Mixtex change their argument passing routine

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: .github/workflows/documentation-windows.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: Documentation Windows
 
 on:
   push:
-    branches: [ master, develop, debug_windows_doc ]
+    branches: [ master, develop, 9818_Windows_documentation ]
```
Comment (jmarrec) [inline .github/workflows/documentation-windows.yml]: Force build this branch, or we won't know until its merged to develop. Running here: https://github.com/NREL/EnergyPlus/actions/runs/4015981786

File: doc/cmake/doc-targets.cmake [hunk 2]
```diff
@@ -2,7 +2,7 @@
 macro( CREATE_DOC_TARGET SOURCE_FILENAME OUTPUT_FILENAME )
   if (NOT UNIX)
     # Avoid issue #8568 - main memory size exceeded for I/O ref on Windows with a recent MikTex (eg 4.1)
-    set(XELATEX_MEM_FLAGS "-extra-mem-top=2000000 -extra-mem-bot=4000000")
+    set(XELATEX_MEM_FLAGS ON)
```
Comment (jmarrec) [inline doc/cmake/doc-targets.cmake:5]: To completely avoid to deal with how to correctly pass a list, I pass a bool, it's clear enough

File: doc/cmake/BuildDocumentation.cmake [hunk 3]
```diff
@@ -15,8 +15,14 @@ if(DOCS_TESTING)
...
   # TODO: You can change this to ON for active debugging if you find a problem
   set(_DEBUG_DOCS OFF)
+  set(COMMAND_ECHO_MODE NONE)
+  if(_DEBUG_DOCS)
+    set(COMMAND_ECHO_MODE STDOUT)
+    set(CMAKE_VERBOSE_MAKEFILE ON)
+  endif()
 
   if (XELATEX_MEM_FLAGS)
+    set(XELATEX_MEM_FLAGS "--extra-mem-top=2000000" "--extra-mem-bot=4000000")
```
Comment (jmarrec) [inline doc/cmake/BuildDocumentation.cmake]: If the bool was passed, make a list of args

File: doc/cmake/BuildDocumentation.cmake [hunk 4]
```diff
@@ -15,8 +15,14 @@ if(DOCS_TESTING)
 
   # TODO: You can change this to ON for active debugging if you find a problem
   set(_DEBUG_DOCS OFF)
+  set(COMMAND_ECHO_MODE NONE)
+  if(_DEBUG_DOCS)
+    set(COMMAND_ECHO_MODE STDOUT)
+    set(CMAKE_VERBOSE_MAKEFILE ON)
+  endif()
```
Comment (jmarrec) [inline doc/cmake/BuildDocumentation.cmake]: improve debugging experience for next time (hopefully never needed)

File: doc/cmake/BuildDocumentation.cmake [hunk 5]
```diff
@@ -74,9 +84,10 @@ if(DOCS_TESTING)
 endif()
 
 execute_process(
-  COMMAND ${XELATEX} -interaction=${THIS_TEX_INTERACTION} ${XELATEX_MEM_FLAGS} ${INNAME}.tex
+  COMMAND "${XELATEX}" --interaction=${THIS_TEX_INTERACTION} ${XELATEX_MEM_FLAGS} ${INNAME}.tex
```
Comment (jmarrec) [inline doc/cmake/BuildDocumentation.cmake:90]: I know xelatex's help shows everything with a single leading `-` but it does understand `--` so I'm using that. 

cf https://tex.stackexchange.com/questions/423161/why-do-both-single-and-double-dash-engine-arguments-work

**Comment by jmarrec** [comment]
https://github.com/NREL/EnergyPlus/actions/runs/4015981786/jobs/6898391204 worked

**Review by Myoldmopar — COMMENTED** [review]
This all seems great to me, just need to remove the temporary branch name from the GHA workflow file.   Once this merges down I'll kick off a dummy package build to make sure all the platforms are happy.

File: .github/workflows/documentation-windows.yml (hunk 1 above)
Comment (jmarrec) [inline .github/workflows/documentation-windows.yml]: Before merging, apply this:

```suggestion
    branches: [ master, develop ]
```

**Comment by jmarrec** [comment]
A full build of a branch that contains this one is currently running at https://github.com/jmarrec/EnergyPlus/actions/runs/4016745063

**Comment by jmarrec** [comment]
@Myoldmopar The full build worked fine. I removed the branch name from the workflow file.
This is ready.

**Comment by Myoldmopar** [comment]
Woo!  Merging before Decent catches it.  Thanks @jmarrec !

**Pull request merged**